src/include/detail/linalg/tdb_matrix_with_ids.h

```diff
@@ -224,19 +224,33 @@ class tdbBlockedMatrixWithIds
 
     auto layout_order = ids_schema_.cell_order();
     this->ids().resize(elements_to_load * dimension);
-    // Create a query
+
+    // Read TileDB data
+    size_t total_size = elements_to_load * dimension;
+    size_t offset = 0;
+    auto ptr = this->ids().data();
     tiledb::Query query(this->ctx_, *ids_array_);
-    query.set_subarray(subarray)
-        .set_layout(layout_order)
-        .set_data_buffer(attr_name, this->ids());
-    tiledb_helpers::submit_query(tdb_func__, ids_uri_, query);
+    query.set_subarray(subarray).set_layout(layout_order);
+    tiledb::Query::Status status;
+    do {
+      // Submit query and get status
+      size_t request_size = READ_BATCH_SIZE;
+      if (offset + READ_BATCH_SIZE > total_size) {
+        request_size = total_size - offset;
+      }
+      query.set_data_buffer(attr_name, ptr + offset, request_size);
+      tiledb_helpers::submit_query(tdb_func__, ids_uri_, query);
+      status = query.query_status();
+
+      auto num_results = query.result_buffer_elements()[attr_name].second;
+      offset += num_results;
+    } while (status == tiledb::Query::Status::INCOMPLETE);
```

Comment:
This needs an additional check for incompletes. You need to check the reason for the incomplete and ensure the buffers are large enough. See the example from [docs](https://docs.tiledb.com/main/how-to/arrays/reading-arrays/incomplete-queries).

Reply:
See here: https://github.com/TileDB-Inc/TileDB-Py/blob/dab850604685c89111a0128729d770ab0650ce24/tiledb/core.cc#L1065-L1069